content/blog/nats-server-2.9.15-release.md

```diff
@@ -0,0 +1,132 @@
+
+The root cause was that the snapshot and compact processes of pull consumer state was triggered as part of the corresponding stream processes. Of course, this would be problematic if the stream is idle!
+
+The fix now decouples these processes so pull consumers will properly snapshot and compact even if there is no new messages in the stream.
+
+**Relevant PRs**
+
+- [#3898](https://github.com/nats-io/nats-server/pull/3898)
+
+## 👷 Hard failures and corruption
+
+Distributed systems are hard, even in the "happy path". However, the real challenge is dealing with abrupt server failures and data corruption. This release brings a plethora of fixes and optimizations for recovering Raft state when these non-ideal failures occur.
+
+This set of fixes are a bit more in the weeds, however, the overarching focus was in more explicit management of state when failures do occur as well as improvements in the Raft implementation to recover a quorum more quickly.
+
+**Relevant PRs**
+
+- [#3911](https://github.com/nats-io/nats-server/pull/3911)
+- [#3922](https://github.com/nats-io/nats-server/pull/3922)
+- [#3934](https://github.com/nats-io/nats-server/pull/3934)
+
+## ❤️‍🩹 Sympathy for the network
+
+When a replica of a stream becomes unavailable for a period of time due to a network partition, server maintenance, etc. once it becomes available, it will need to “catch-up” with the leader. A related scenario is when a stream is scaled up from one replica to multiple (say three), these new replicas need the data.
+
+Each replica signals to the leader it has stream data up to a particular sequence (zero if new) and the leader will asynchronously send all of the messages beyond that point to the replica so it can commit it locally.
+
+For busy streams, even a small amount of time being unavailable could result in a large amount of data being sent in order to catch-up. Now consider that when a replica becomes unavailable, it is often due to an entire server in the cluster being offline. Since a server can host many replicas in the cluster, this automated catch-up process can exacerbate the effect.
+
+All of this catch-up takes up bandwidth and, in the worst case, could completely saturate the network causing throughput of the client applications to drop. Fortunately, there is a little known configuration option on the server called `jetstream.max_outstanding_catchup` which denotes the amount of data that is allowed to be outstanding (in-flight) at any given time during this catch-up process.
+
+The previous default value was 128MB and was chosen with the assumption that 10Gbit links (or better) would be used between servers in a cluster. As a result of user-reported issues of “slow consumers” between servers during this catch-up process and a series of I/O-heavy stress tests, the default value has been reduced to 64MB.
```

Comment:
Do we need the 'it' on 10Gbit? I thought it would just be 10GB?

Reply:
They are different units. GB means "gigabyte" and Gbit means "gigabit". There are eight bits in a byte. Often interchanged/confused especially with internet service providers.

Reply:
I could do `Gb`, but that could still be interpreted to be bytes.. so I choose to be explicit with `Gbit`.